src/os/bluestore/BlueStore.cc

```diff
@@ -12709,10 +12729,27 @@ int BlueStore::queue_transactions(
   OpSequencer *osr = c->osr.get();
   dout(10) << __func__ << " ch " << c << " " << c->cid << dendl;
 
+#ifdef HAVE_JAEGER
+  jspan span = nullptr;
+  if (op && op->osd_parent_span) {
+    span = move(jaeger_tracing::child_span("BlueStore::queue_transactions",
+                                           op->osd_parent_span));
+  }
+#endif
+
   // prepare
   TransContext *txc = _txc_create(static_cast<Collection*>(ch.get()), osr,
 				  &on_commit, op);
 
+#ifdef HAVE_JAEGER
+  if (span) {
```

Comment:
Is there some reason - like time stamp capture - that justifies creation of child_span before creation of txc?

Reply:
@aclamk PR is reconstructed now to make the code easy to maintain